src/generic_ui/scripts/ui.ts

```diff
@@ -968,19 +968,42 @@ export class UserInterface implements ui_constants.UiApi {
     });
   }
 
+
   public logout = (networkInfo :social.SocialNetworkInfo) : Promise<void> => {
     var network = this.model.getNetwork(networkInfo.name);
+
+    //If the user is connecteed to a network, do a check to see if the user is sure that they want to log out
     if (network) {
-      // if we know about the network, record that we expect this logout to
-      // happen
-      network.logoutExpected = true;
+      var instanceIds = Object.keys(this.instancesGivingAccessTo);
+      var confirmationMessage: string;
+      if (instanceIds.length > 0) {
+
+        confirmationMessage = this.i18n_t("PRE_LOG_OUT", {
+          name: this.mapInstanceIdToUser_[instanceIds[0]].name,
+          numOthers: (instanceIds.length - 1)
+        });
+      }
+      else {
```

Comment:
The else statement should be on the same line as the closing brace (combine lines 985 and 986)